osquery/tables/system/windows/windows_search.cpp

```diff
@@ -0,0 +1,386 @@
+      wss << variant.hVal.HighPart << variant.hVal.LowPart;
+      break;
+    case VT_DATE:
+      wss << dateToUnixTime(variant.date);
+      break;
+    default:
+      wss << "unhandled variant type " << variant.vt;
+      break;
+  }
+}
+
+std::string ccomandColumnStringValue(CCommand<CDynamicAccessor, CRowset>& cCommand, DBORDINAL columnIndex) {
+  DBTYPE type;
+  cCommand.GetColumnType(columnIndex, &type);
+
+  std::wstringstream wss;
+
+  DBSTATUS status;
+  cCommand.GetStatus(columnIndex, &status);
+  switch (status) {
+    case DBSTATUS_S_ISNULL: {
+      wss << "NULL";
+      break;
+    }
+    case DBSTATUS_S_OK:
+    case DBSTATUS_S_TRUNCATED: {
+      switch (type) {
+        case DBTYPE_VARIANT: {
+          writePropVariant(*(static_cast<PROPVARIANT*>(cCommand.GetValue(columnIndex))), wss);
+          break;
+        }
+        case DBTYPE_WSTR: {
+          DBLENGTH cbLen;
+          cCommand.GetLength(columnIndex, &cbLen);
+          WCHAR szBuffer[2048];
+          StringCchCopyN(szBuffer, ARRAYSIZE(szBuffer), static_cast<WCHAR*>(cCommand.GetValue(columnIndex)), cbLen / sizeof(WCHAR));
+          wss << szBuffer;
+          break;
+        }
+        case DBTYPE_I1: {
+          wss << *static_cast<UCHAR*>(cCommand.GetValue(columnIndex));
+          break;
+        }
+        case DBTYPE_UI2: {
+          wss << *static_cast<USHORT*>(cCommand.GetValue(columnIndex));
+          break;
+        }
+        case DBTYPE_I2: {
+          wss << *static_cast<SHORT*>(cCommand.GetValue(columnIndex));
+          break;
+        }
+        case DBTYPE_UI4: {
+          wss << *static_cast<DWORD*>(cCommand.GetValue(columnIndex));
+          break;
+        }
+        case DBTYPE_I4: {
+          wss << *static_cast<INT*>(cCommand.GetValue(columnIndex));
+          break;
+        }
+        case DBTYPE_UI8: {
+          wss << *static_cast<ULONGLONG*>(cCommand.GetValue(columnIndex));
+          break;
+        }
+        case DBTYPE_I8: {
+          wss << *static_cast<LONGLONG*>(cCommand.GetValue(columnIndex));
+          break;
+        }
+        case DBTYPE_DATE: {
+          wss << dateToUnixTime(*static_cast<DATE*>(cCommand.GetValue(columnIndex)));
+          break;
+        }
+        default: {
+          wss << "unhandled database type " << type;
+          break;
+        }
+      }
+      break;
+    }
+    default: {
+      wss << "error reading column";
+      break;
+    }
+  }
+
+  std::wstring wideStr = wss.str();
+  std::wstring_convert<std::codecvt_utf8<wchar_t>> converter;
+  return converter.to_bytes(wideStr);
+}
+
+std::vector<std::map<std::string, std::string>> executeWindowsSearchQuery(CSession& cSession, const std::string& query) {
+  HRESULT hr = NULL;
+  std::vector<std::map<std::string, std::string>> results;
+
+  CCommand<CDynamicAccessor, CRowset> cCommand;
+  hr = cCommand.Open(cSession, query.c_str());
+
+  if (FAILED(hr)) {
+    LOG(ERROR) << "error executing query";
+    return results;
+  }
```

Comment:
Do we need to `cCommand.Close()` here? Should that be in a scope guard? (like a go defer)

Reply:
scope guarded